examples/migration/recursive_fibonacci/fibonacci.cpp

```diff
@@ -0,0 +1,100 @@
+/*
+    Copyright (c) 2023 Intel Corporation
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+#include "task_emulation_layer.h"
+
+#include <iostream>
+#include <numeric>
+#include <utility>
+
+int cutoff;
+
+long serial_fib(int n) {
+    return n < 2 ? n : serial_fib(n - 1) + serial_fib(n - 2);
+}
+
+struct fib_continuation : task::base_task {
+    fib_continuation(int& s) : sum(s) {}
+
+    void execute() override {
+        sum = x + y;
+    }
+
+    int x{ 0 }, y{ 0 };
+    int& sum;
+};
+
+struct fib_computation : task::base_task {
+    fib_computation(int n, int* x) : n(n), x(x) {}
+
+    void execute() override {
+        if (n < cutoff) {
+            *x = serial_fib(n);
+        }
+        else {
+            // Continuation passing
+            auto& c = *this->create_continuation<fib_continuation>(/* children_counter = */ 2, *x);
+            task::run_task(c.create_child_of_continuation<fib_computation>(n - 1, &c.x));
+
+            // Recycling
+            this->recycle_as_child_of_continuation(c);
+            n = n - 2;
+            x = &c.y;
+
+            // Bypass
```

Comment:
Maybe extend this comment to clarify how this differs old-TBB bypass, that is, that you do not return the next task from the function but instead execute it directly here.  You can mention that it is possible with "defer" if necessary, but is not needed for this example.

Reply:
Changed the comment.
The problem is that this emulation layer does not support `defer` so TBB-like bypass is not possible here.